docs/docs/dev/ci.md

```diff
@@ -0,0 +1 @@
+# CI
```

Comment:
I guess this is still WIP?

Reply:
It was, I added some to it now.
I wouldn't document more than the action I added now in this PR. WDYT?

Reply:
The documentation is a bit overkill. A brief explanation of how the workflows are working is completely sufficient.
We don't want to reuse these actions in other repositories. The documentation should only inform the developers how we setup the CI/CD

Reply:
It is also outdated now, I have deleted the action entirely. Should I document anything at all? Seems a bit out of scope for this PR.

Perhaps just the added hook and step in the CI?

Reply:
Yeah, let's just skip it for this PR.

Reply:
okay, removed the doc